utils/utils.go

```diff
@@ -182,3 +189,44 @@ func ToJsonOrDie(value interface{}) []byte {
 		panic(err)
 	}
 }
+
+func DoesCRDExist(crdName string) bool {
+
+	var options client.Options
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(apiextensionsv1.AddToScheme(options.Scheme))
+
+	setupLog := ctrl.Log.WithName("setup")
+
+	k8sClient, err := client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		setupLog.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	crd := apiextensionsv1.CustomResourceDefinitionList{}
+	listOptions := &client.ListOptions{
+		FieldSelector: fields.OneTermEqualSelector("metadata.name", crdName),
+	}
+
+	err = k8sClient.List(context.Background(), &crd, listOptions)
+
+	if err != nil {
+		setupLog.Error(err, fmt.Sprintf("Error Getting CRD for %s", crdName))
+		return false
+	} else if len(crd.Items) == 0 {
+		return false
+	} else if crd.Items[0].Name == crdName {
+		return true
+	}
+	return false
+}
```

Comment:
This entire function can be replaced with a single `r.Get` on the call sites.  Is there any reason for having it and doing the existence check this way?